google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -208,7 +208,8 @@ def create(
             dataset (Dataset):
                 Instantiated representation of the managed dataset resource.
         """
-
+        if not display_name:
```

Comment:
We should explicitly check if display_name is `None` to not override in the case of an empty string.

Reply:
Note that resources with empty display name are very hard to use in Pantheon.
In Pantheon the user must click on the resource name in the first column. But that's impossible when the name is empty.
I've logged a bug about this, but I'm not sure whether it's going to be fixed since it's a common Pantheon UX control.

Reply:
Got it, thanks for the context. I agree with your approach.